data/src/main/java/com/microsoft/azure/kusto/data/AadAuthenticationHelper.java

```diff
@@ -148,4 +150,16 @@ AuthenticationResult acquireAccessTokenUsingDeviceCodeFlow() throws Exception {
         return result;
     }
 
+    public AuthenticationResult acquireWithClientCertificate(X509Certificate cert, PrivateKey privateKey)
+            throws IOException, InterruptedException, ExecutionException{
+
+        AuthenticationContext context = new AuthenticationContext(aadAuthorityUri, false, Executors.newFixedThreadPool(1));
```

Comment:
Instead of passing Executors.newFixedThreadPool(1) in second parameter, you might initiate in try{} block: 
service = Executors.newSingleThreadExecutor();
and then in finally shut it down: (as did in acquireAccessTokenUsingDeviceCodeFlow())
} finally {
            if (service != null) {
                service.shutdown();
            }
        }